### PR TITLE
[FEAT] Adding eslint-plugin-qunit

### PR DESCRIPTION
**Pull request by pete-the-pete**
This addresses #6090 by adding the eslint plugin. Event though the actual lint rule doesn't enforce having a label for assertions (https://github.com/platinumazure/eslint-plugin-qunit/blob/master/docs/rules/assert-args.md) I feel like this could still be worthwhile to have.

### CODE REVIEW

**Comment by igorT** [comment]
Looks good to me. Seems like it ran for existing files though?

**Comment by pete-the-pete** [comment]
oops, not quite there...I'll update.

File: bin/lint-js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+if (LIST) {
+  LIST = LIST.split(/\r?\n/);
+  // filter for javascript
+  LIST = LIST.filter(item => item.match(/^[^.].*\.js$/)).join();
+  console.log("LIST LENGTH", LIST.length)
+  if (LIST) {
+    try {
+      //update eslint config to include qunit
+      eslintInfo.extends.push('plugin:qunit/recommended');
```
Comment (hjdivad) [inline bin/lint-js]: What's the reason for adding the config here and not in `.eslintrc.js`?

**Comment by pete-the-pete** [comment]
As it currently stands, running qunit eslint raises a lot of issues. Adding
it to the main .eslintrc.js means the qunit rules will in charles/azure,
and things start failing. I couldn't find a way to have only 1 plugin run
at a time, so manipulating the config at runtime was the workaround i came
up with.

If/When the qunit eslint issues have been dealt with we can remove this and
add it to the main eslintrc.

**Pull request merged**